tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -465,3 +465,42 @@ def update_logging(storage_type, storage_account, logging_settings, session=None
 
         return getattr(client, 'set_{}_service_properties'
                        .format(storage_type))(logging=logging_settings)
+
+
+@Storage.action_registry.register('set-https-traffic')
+class SetHttpsTrafficAction(AzureBaseAction):
+    """Action that updates the EnableHttpsTrafficOnly setting on Storage Accounts.
+
+    :example:
+
+        Set EnableHttpsTrafficOnly to True
+
+    .. code-block:: yaml
+
+        policies:
+            - name: set-https-traffic-to-true
+              resource: azure.storage
+              actions:
+              - type: set-https-traffic
+                value: true
```

Comment:
Maybe 

```
-type: require-secure-transfer
 enabled: true
```

This is the end-user facing terminology:
https://docs.microsoft.com/en-us/azure/storage/common/storage-require-secure-transfer